client/web/src/components/views/readonly-client-view.tsx

```diff
@@ -23,6 +23,25 @@ export default function ReadonlyClientView({
   auth?: AuthResponse
   sessions: SessionsCallbacks
 }) {
+  const newSession = async () => {
+    return sessions
+      .new()
+      .then((url) => {
+        if (url) {
+          window.open(url, "_blank")
+          sessions.wait()
+        }
+      })
+      .catch((e) => console.error(e))
+  }
+
+  useEffect(() => {
+    // if ?checkmode=1 is provided, get a new session immediately
+    if (new URLSearchParams(window.location.search).get("checkmode") == "1") {
```

Comment:
Could move `new URLSearchParams(window.location.search).get("checkmode") == "1"` into a little helper func.

This whole `useEffect` could also be moved into `useAuth`

Reply:
moved into the existing `useEffect` in `useAuth`